File: run_forecast.R
args <- commandArgs(trailingOnly = TRUE)

get_arg <- function(name, default = NULL) {
  pattern <- paste0("--", name, "=")
  hit <- args[grepl(pattern, args)]
  if (length(hit) == 0) return(default)
  sub(pattern, "", hit[1])
}

source("R/forecast_module.R")

data_path <- get_arg("data", "data.xlsx")
artifacts_dir <- get_arg("artifacts", "artifacts")
output_path <- get_arg("output", "artifacts/imae_forecast_12m.csv")
horizon <- as.integer(get_arg("horizon", "12"))

forecast_df <- generate_imae_projection(
  data_path = data_path,
  artifacts_dir = artifacts_dir,
  horizon = horizon,
  output_path = output_path
)

print(forecast_df)

File: run_forecast.py
from __future__ import annotations

import argparse

from imae_forecasting.forecasting import generate_12m_forecast


def main() -> None:
    parser = argparse.ArgumentParser(description="Uso del modelo para proyección IMAE")
    parser.add_argument("--data", default="data.xlsx", help="Ruta al archivo Excel de entrada")
    parser.add_argument("--artifacts", default="artifacts", help="Directorio con modelo entrenado")
    parser.add_argument("--output", default="artifacts/imae_forecast_12m.csv", help="CSV de proyección")
    parser.add_argument("--horizon", default=12, type=int, help="Horizonte de proyección")
    args = parser.parse_args()

    forecast = generate_12m_forecast(
        data_path=args.data,
        artifacts_dir=args.artifacts,
        output_path=args.output,
        horizon=args.horizon,
    )
    print(forecast[["imae_proyectado"]].tail(args.horizon))


if __name__ == "__main__":
    main()
